index.html

```diff
@@ -310,6 +293,31 @@ <h2>The <dfn>PerformanceMark</dfn> Interface</h2>
       <p>The <code>duration</code> attribute must return a <a data-cite="!HR-TIME-2#idl-def-domhighrestimestamp"><code>DOMHighResTimeStamp</code></a> of value <code>0</code>.</p>
       <p>The <a>PerformanceMark</a> interface contains the following additional attribute:</p>
       <p>The <dfn>detail</dfn> attribute must return the value it is set to (it's copied from the <a>PerformanceMarkOptions</a> dictionary).</p>
+      <section data-link-for="PerformanceMarkOptions">
+        <h3>The <dfn><a>PerformanceMark</a> Constructor</dfn></h3>
+        <p>The <a>PerformanceMark constructor</a> must run the following steps:</p>
+        <ol>
+          <li>If the <a data-cite="!HTML51/webappapis.html#global-object">global object</a> is a <code>Window</code> object and <var>markName</var> uses the same name as a <a data-cite="!WEBIDL#dfn-read-only">read only attribute</a> in the <code><a data-cite="!NAVIGATION-TIMING#performancetiming">PerformanceTiming</a></code> interface, <a data-cite="!WEBIDL#dfn-throw">throw</a> a <a data-cite="!WEBIDL#syntaxerror"><code>SyntaxError</code></a>.</li>
+          <li>Create a new <a>PerformanceMark</a> object (<var>entry</var>).</li>
+          <li>Set <var>entry</var>'s <code>name</code> attribute to <var>markName</var>.</li>
+          <li>Set <var>entry</var>'s <code>entryType</code> attribute to <code>DOMString "mark"</code>.</li>
+          <li>
+            Set <var>entry</var>'s <code>startTime</code> attribute as follows:
+            <ol>
+              <li>If <var>markOptions</var> is present and its <a>startTime</a> member is present, then set it to the value of <var>markOptions</var>'s <a>startTime</a>.</li>

```

Comment:
What kind of validation are you hoping for? The idea of UserTiming L3 is to allow arbitrary timestamps, so I think technically any number used would be valid.

Reply:
- Negative values are OK? :)
- Floats?
- What happens if I pass in a non-numeric value?

I expect at least some of those to raise an exception when I try to pass them to the constructor?

Reply:
Negative and otherwise too low values could be considered invalid, maybe worth discussing how this would be useful and which timestamps should be considered invalid? I'd be ok adding some checks to make sure the timestamp used is reasonable.

Floats are of course valid as DOMHighResTimeStamp is double.

I think we do not need to check for non-numeric values manually? By specifying the type in the dictionary, we ensure that the implementor will throw a TypeError whenever the value passed cannot be converted to that type.